spec/tasks/update_history.Tests.ps1

```diff
@@ -0,0 +1,159 @@
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+$sut = (Split-Path -Leaf $MyInvocation.MyCommand.Path).Replace(".Tests.", ".")
+$helper = Join-Path (Split-Path -Parent $here) 'spec_helper.ps1'
+. $helper
+$sut = Join-Path -Path $src -ChildPath "tasks/${sut}"
+
+. $sut -NoOperation
+
+Describe 'Convert-ToServerSelectionString' {
+  It 'should enumerate default server' {
+    Convert-ToServerSelectionString 0 | Should Be 'Default'
+  }
+
+  It 'should enumerate managed server' {
+    Convert-ToServerSelectionString 1 | Should Be 'ManagedServer'
+  }
+
+  It 'should enumerate a Windows Update server' {
+    Convert-ToServerSelectionString 2 | Should Be 'WindowsUpdate'
+  }
+
+  It 'should enumerate an other server' {
+    Convert-ToServerSelectionString 3 | Should Be 'Other'
+  }
+}
+
+Describe 'Convert-ToOperationResultCodeString' {
+  It 'should enumerate not started operation' {
+    Convert-ToOperationResultCodeString 0 | Should Be 'Not Started'
+  }
+
+  It 'should enumerate in progress operation' {
+    Convert-ToOperationResultCodeString 1 | Should Be 'In Progress'
+  }
+
+  It 'should enumerate a succesful operation' {
+    Convert-ToOperationResultCodeString 2 | Should Be 'Succeeded'
+  }
+
+  It 'should enumerate a succesful with errors operation' {
+    Convert-ToOperationResultCodeString 3 | Should Be 'Succeeded With Errors'
+  }
+
+  It 'should enumerate a failed operation' {
+    Convert-ToOperationResultCodeString 4 | Should Be 'Failed'
+  }
+
+  It 'should enumerate an abort operation' {
+    Convert-ToOperationResultCodeString 5 | Should Be 'Aborted'
+  }
+}
+
+Describe 'Convert-ToUpdateOperationString' {
+  It 'should enumerate installation operations' {
+    Convert-ToUpdateOperationString 1 | Should Be 'Installation'
+  }
+
+  It 'should enumerate uninstallation operations' {
+    Convert-ToUpdateOperationString 2 | Should Be 'Uninstallation'
+  }
+}
+
+Describe 'Invoke-ExecuteTask' {
+  $DefaultExecuteParams = @{
+    Detailed = $false;
+    Title = $null
+    UpdateID = $null
+    MaximumUpdates = 300
+  }
+
+  It 'should return empty JSON if no history' {
+    Mock Get-UpdateSessionObject { New-MockUpdateSession 0 }
+
+    $Result = Invoke-ExecuteTask @DefaultExecuteParams | ConvertFrom-JSON
+    $Result.Count | Should be 0
+  }
+
+  It 'should return a JSON array for a single element' {
+    Mock Get-UpdateSessionObject { New-MockUpdateSession 1 }
+
+    $Result = Invoke-ExecuteTask @DefaultExecuteParams
+    $Result | Should -Match "^\[ "
+    $Result | Should -Match " \]$"
```

Comment:
I've changed the test slightly now.  It now checks that the JSON string starts and stops with brackets AND that it can be converted back into a PowerShell object array with a single element.  Why I didn't use a literal string is that the `Match` doesn't do multiline and all I really care is that it's a JSON array.